itest/lnd_delete_canceled_invoice_test.go

```diff
@@ -0,0 +1,93 @@
+package itest
+
+import (
+	"bytes"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/invoices"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/stretchr/testify/require"
+)
+
+func testDeleteCanceledInvoice(ht *lntest.HarnessTest) {
+	// Open a channel with 100k satoshis between Alice and Bob.
```

Comment:
I think we don't need a network here only if you also want to settle an invoice which you don't do in the test, so either also test that you cannot cancel a settled invoice or just create a node without Alice. Bob but only Alice.